networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/BeaconChainMethods.java

```diff
@@ -242,9 +223,10 @@ private static Eth2RpcMethod<GoodbyeMessage, GoodbyeMessage> createGoodBye(
                   peerLookup);
 
       return VersionedEth2RpcMethod.create(
-          rpcEncoding, requestType, expectResponseToRequest, List.of(v2Method, v1Method));
+          rpcEncoding, requestType, expectResponseToRequest, List.of(v2Method));
     } else {
-      return v1Method;
+      return VersionedEth2RpcMethod.create(
+          rpcEncoding, requestType, expectResponseToRequest, List.of());
```

Comment:
The method list should not be empty here.
I guess there is not need to check the milestone anymore. We could directly return a `VersionedEth2RpcMethod` with the `v2Method`.
I tried that and it fixed most of the integrations tests btw

Reply:
Okay, I could do that. Is that the right approach? The [spec](https://github.com/ethereum/consensus-specs/pull/3014/files) says "Clients MAY respond with an empty list during the deprecation transition period." If that is not what we want to do, I can try your suggestions returning the `v2Method`?

Reply:
In my understanding, that means that both `beacon_blocks_by_range` and `beacon_blocks_by_root` may return an empty list of blocks which means implementing that behaviour in the v1methods before completely removing them